packages/database/src/modelClasses/Entity.js

```diff
@@ -237,9 +238,8 @@ export class EntityModel extends DatabaseModel {
     );
   }
 
-  async fetchAncestorDetailsByDescendantCode(...args) {
-    const cacheKey = `fetchAncestorDetailsByDescendantCode:${JSON.stringify(args)}`;
-    const [descendantCodes, hierarchyId, ancestorType] = args;
+  async fetchAncestorDetailsByDescendantCode(descendantCodes, hierarchyId, ancestorType) {
+    const cacheKey = this.getCacheKey('fetchAncestorDetailsByDescendantCode', arguments);
```

Comment:
Optional change, perhaps it makes this a bit safer:

```suggestion
    const cacheKey = this.getCacheKey(this.fetchAncestorDetailsByDescendantCode.name, arguments);
```

Reply:
Yeah I like this